pinot-spi/src/main/java/org/apache/pinot/spi/env/PinotConfiguration.java

```diff
@@ -201,16 +200,16 @@ private static Configuration loadProperties(String configPath) {
   private static Map<String, Object> relaxConfigurationKeys(Configuration configuration) {
     return CommonsConfigurationUtils.getKeysStream(configuration).distinct()
 
-        .collect(Collectors.toMap(PinotConfiguration::relaxPropertyName, key -> configuration.getProperty(key)));
+        .collect(Collectors.toMap(PinotConfiguration::relaxPropertyName, configuration::getProperty));
   }
 
   private static Map<String, String> relaxEnvironmentVariables(Map<String, String> environmentVariables) {
-    return environmentVariables.entrySet().stream().filter(entry -> entry.getKey().startsWith("PINOT_"))
+    return environmentVariables.entrySet().stream().filter(entry -> entry.getKey().startsWith(ENV_PREFIX))
```

Comment:
This will break all existing changes right?

I would suggest to keep current code rename it to legacy, so current user won't fail for using `PINOT_` prefix. 

Then add new code to override based on the new prefix.

Reply:
At the moment (before and after my changes) `<PREFIX>_SERVER_xxx` is mapped to `server.xxx`. I've just changed the prefix to avoid confusion since `pinot.server.property` will be loaded from `PINOT_PINOT_SERVER_PROPERTY`. Considering that env wasn't loaded until now I think there's no worries about backward compatibility.

Reply:
Moreover I think it would be quite hard to enforce an identical prefix to all properties. I'm for having a custom self-explanatory prefix like `PINOT_ENV_`. But I'm open to better ideas.

Reply:
1.
For the backward compatible perspective, I think the issue here is that for whoever currently using `PINOT_CONTROLLER_HOST` will see the unexpected behavior the override won't happen.

Suggest to rename existing `relaxEnvVarName` to `legacyRelaxEnvVarName`, and have your new `relaxEnvVarName` to override.

E.g. existing code:
`PINOT_CONTROLLER_HOST` should be added as `controller.host`;
`PINOT_PINOT_CONTROLLER_HOST` should be added as `pinot.controller.host`;

`PINOT_ENV_SERVER_XXX` should be added as `server.xxx`;
`PINOT_ENV_PINOT_SERVER_XXX` should be added as `pinot.server.xxx`;

2.
From the ease of use perspective, actually I would love to have the direct mapping from env variable to configs, e.g.
`pinot.server.xxx` should be override by `PINOT_SERVER_XXX`.
But due to some legacy issue that not all configs are start with `pinot.` that's why we introduce this `PINOT_` prefix.

IMO, I would love to map env var `PINOT_XXX_YYY` to both `xxx.yyy` and `pinot.xxx.yyy`.

So that we can cover both.

Just for `xxx.yyy` may override existing configs but `pinot.xxx.yyy` will be added if not existed.

cc: @Jackie-Jiang @snleee @siddharthteotia @chenboat would love your best practice of config override.

Reply:
Implemented 1.
I'm worried that doing 2. we could incur in conflicts with existing env vars. Prefix avoids this issues.

Reply:
After catching up, number two was implemented following:
```
export PINOT_CONTROLLER_HOST=host
export PINOT_SERVER_PROPERTY_WHATEVER=whatever_property
export ANOTHER_VARIABLE=random

-------------------- Config from CLI or Config Path ------------------------

dynamic.env.config=pinot.controller.host,pinot.server.property,other.var
pinot.controller.host=PINOT_CONTROLLER_HOST
pinot.server.property=PINOT_SERVER_PROPERTY_WHATEVER
other.var=ANOTHER_VARIABLE

---------------------------------------------------------------------------

config gets templated at runtime with the content of the env vars.

----------------------- final result in memory ----------------------------

dynamic.env.config=pinot.controller.host,pinot.server.property,other.var
pinot.controller.host=host
pinot.server.property=whatever_property
other.var=random
```